### ISSUE TITLE
Implement k-layer neural network

### ISSUE THREAD

**Issue by tjr16** [tjr16/neural-networks#2]

**Comment by tjr16** [comment]
TODO: exponential moving average for batch normalization
